### PR TITLE
Harvest Temple cleanup and fixes, Food damage modifiers

### PR DESCRIPTION
**Pull request by Linkaaaaa**
- Fixed Primordus' Jaws of Destruction circle was mispositioned and too large on the sides
- Fixed Primordus' Jaws of Destruction damage field was spawning too late
- Fixed Primordus' Jaws of Destruction indicator and damage field wouldn't spawn due to an encounter bug
- Fixed Primordus' Jaws of Destruction and Lava Slam wouldn't disappear when phasing
- Fixed Primordus' Lava Slam position
- Fixed Zhaitan's Putrid Deluge AoEs not spawning correctly if the aoes appear after the target last aware
- Fixed Kralkatorrik's Beam Pools lasting shorter than intended, players were damaged by an invisible effect
- Fixed Pool of Undeath lasting shorter than intended and improved their display
- Fixed Stormseer' Ice Spikes being under Jormag instead of Environment (effect has no Src)
- Fixed Stormseer' Ice Spikes indicator lasting too long
- Adjusted Zhaitan's Reach inner doughnut radius
- Adjusted Saltspray's Shatter Earth doughnuts radiuses
- Adjusted Saltspray's Frozen Fury cone radius and removed offset for the rectangles
- Updated Magic Discharge to be a shockwave
- Added Grasp of Jormag AoEs
- Added circles to Jormag's Frost Beam
- General cleanup to decorations' lifespans and colors

- Added food damage modifiers (addictive and multiplicative lists)

### CODE REVIEW

File: GW2EIEvtcParser/EncounterLogic/Strikes/EndOfDragons/HarvestTemple.cs [hunk 1]
```diff
@@ -977,18 +1031,25 @@ internal override void ComputeNPCCombatReplayActors(NPC target, ParsedEvtcLog lo
...
                         foreach (EffectEvent effect in zhaitanPoisonAoeEffects)
                         {
-                            int start = (int)effect.Time;
-                            int end = (int)Math.Min(target.LastAware, start + 10000);
-                            replay.AddDecorationWithBorder(new CircleDecoration(200, (start, end), "rgba(49, 71, 0, 0.2)", new PositionConnector(effect.Position)), "rgba(200, 0, 0, 0.5)");
+                            int fieldDuration = 10000;
+                            (long start, long end) lifespan = (effect.Time, effect.Time + fieldDuration);
+                            if (lifespan.start > target.LastAware)
+                            {
+                                lifespan.end = effect.Time + 10000;
```
Comment (EliphasNUIT) [inline GW2EIEvtcParser/EncounterLogic/Strikes/EndOfDragons/HarvestTemple.cs:1046]: We only need the else no? Also those + 10000 should be + fieldDuration

File: GW2EIEvtcParser/EncounterLogic/Strikes/EndOfDragons/HarvestTemple.cs [hunk 2]
```diff
@@ -876,20 +932,19 @@ internal override void ComputeNPCCombatReplayActors(NPC target, ParsedEvtcLog lo
                         foreach (EffectEvent effect in kralkBeamEffects)
                         {
                             int indicatorDuration = 2000;
-                            int aoeDuration = 5000;
-                            int indicatorStart = (int)effect.Time;
-                            int aoeStart = indicatorStart + indicatorDuration;
-                            int aoeEnd = Math.Min(aoeStart + aoeDuration, (int)target.LastAware);
-                            replay.AddDecorationWithGrowing(new RectangleDecoration(700, 2900, (indicatorStart, aoeEnd), Colors.Orange, 0.2, new PositionConnector(effect.Position)), aoeStart);
+                            long growingEnd = effect.Time + indicatorDuration;
```
Comment (EliphasNUIT) [inline GW2EIEvtcParser/EncounterLogic/Strikes/EndOfDragons/HarvestTemple.cs:935]: The lifespan of the rectangle has been reduced, intended?

Comment (Linkaaaaa) [reply GW2EIEvtcParser/EncounterLogic/Strikes/EndOfDragons/HarvestTemple.cs:935]: Yeah I've changed it to appear only as indicator just like in game, having the rectangle shown as dangerous in spots where there's no damage isn't ideal imo

File: GW2EIEvtcParser/EncounterLogic/Strikes/EndOfDragons/HarvestTemple.cs [hunk 3]
```diff
@@ -829,44 +840,89 @@ internal override void ComputeNPCCombatReplayActors(NPC target, ParsedEvtcLog lo
...
+                            lifespan.end = Math.Min(lifespan.end, target.LastAware);
+                            replay.AddDecorationWithGrowing(new CircleDecoration(580, lifespan, Colors.Orange, 0.2, new PositionConnector(effect.Position)), growingEnd);
+                        }
+                    }
+
+                    // We use the damage field for 2 reasons:
+                    // 1: When the instance is bugged from a previous wipe, the orange warning indicator is not present but the red damage field is always visible.
+                    // 2: The red damage field is not in the same position of the orange warning indicator, rending the indicator inaccurate.
+                    
+                    var jawsOfDestructionPoint = new Point3D(591.0542f, -21528.8555f, -15418.3f);
```
Comment (EliphasNUIT) [inline GW2EIEvtcParser/EncounterLogic/Strikes/EndOfDragons/HarvestTemple.cs:892]: renamed to xxxPosition and the one below to xxxConnector

File: GW2EIEvtcParser/EncounterLogic/Strikes/EndOfDragons/HarvestTemple.cs (hunk 1 above)
Comment (Linkaaaaa) [reply GW2EIEvtcParser/EncounterLogic/Strikes/EndOfDragons/HarvestTemple.cs:1046]: Yeah can invert the if to be lifespan.Start < target.LastAware and only have the content of the else

File: GW2EIEvtcParser/EIData/DamageModifiers/CommonDamageModifiers/ItemDamageModifiers.cs [hunk 4]
```diff
@@ -25,6 +25,21 @@ internal static class ItemDamageModifiers
...
+            new BuffOnActorDamageModifier(CanOfStewedOysters, "Food: -8%", "-8%", DamageSource.All, -8.0, DamageType.Strike, DamageType.All, Source.Item, ByPresence, BuffImages.NourishmentFood, DamageModifierMode.PvEWvW),
+            new BuffOnActorDamageModifier(
+                new long[]
+                {
+                    PlateOfPeppercornSpicedCoqAuVin, MistsInfusedPeppercornCrustedSousVideSteak, PeppercornCrustedSousVideSteak, PlateOfPepperedClearTruffleRavioli,
+                    PlateOfPeppercornSpicedBeefCarpaccio, PlateOfPeppercornSpicedPoultryAspic, PeppercornSpicedEggsBenedict, SpicedPepperCremeBrulee,
+                    SpicedPeppercornCheesecake, PepperedCuredMeatFlatbread, PeppercornAndVeggieFlatbread, BowlOfSpicedFruitSalad, 
+                    SpherifiedPeppercornSpicedOysterSoup, MistsInfusedSpherifiedPeppercornSpicedOysterSoup,
+                }, "Ascended Food: -10%", "-10%", DamageSource.NoPets, -10.0, DamageType.Strike, DamageType.All, Source.Item, ByMultiPresence, BuffImages.NourishmentFood, DamageModifierMode.PvEWvW),
```
Comment (EliphasNUIT) [inline GW2EIEvtcParser/EIData/DamageModifiers/CommonDamageModifiers/ItemDamageModifiers.cs:36]: ByPresence, although impossible, ByMultiPresence will stack the bonus per present nourishment on the player

**Comment by EliphasNUIT** [comment]
Thanks!

**Pull request merged**